### PR TITLE
added cob_collision_monitor package

### PR DESCRIPTION
**Pull request by mathias-luedtke**
Collision monitor node and capability that outputs ~state_is_valid (std_msgs/Bool) topic with current  state validity.
It publishes true until the state of all joints is known.
Afterwards if checks the age of the joint value and the validity of the current state.
A ground plane/box is added automatically.

Both take private parameters:
- max_age(float, default: 1.0): maximum age for complete joint state
- verbose (bool, default: true): print collision information
- add_ground (bool, default: true): add ground plane/box
- ground_link (string, default: base_link): frame for ground plane/box
- ground_height (float, default: -0.001): ground level
- ground_box_size (float): use box of given size instead of plane (https://github.com/flexible-collision-library/fcl/issues/12)

The node takes additional parameters:
- robot_description (string): robot description ;)
- max_cache_time(float, default: 2.0): maximum cache time for transforms
- state_update_frequency(float): set rate limiter if given (moveit defaults to 10 Hz)
- start_scene_monitor (bool, default: true): start scene monitor
- start_world_geometry_monitor (bool, default: true): start world geometry monitor

### CODE REVIEW

**Comment by fmessmer** [comment]
Check out [cob_obstacle_distance](https://github.com/ipa320/cob_control/tree/indigo_dev/cob_obstacle_distance)
It is significantly improved in https://github.com/ipa320/cob_control/pull/46 and it is still undergoing more changes....

@ipa-fxm-mb uses it for (self-)collision-avoidance strategies in cob_twist_controller...it provides you with a distance value to the closest obstacle...and it does not require moveit to be started....

Maybe synergies can be found and applied!

**Comment by mathias-luedtke** [comment]
This node should be used to stop on self collisions for all joints, especially for dashboard commands.
It does not require the move_group node, it can be used stand-alone (with an dedicated planning scene monitor under the hood).

I just had a brief look at cob_obstacle_distance, it looks like a reimplementation of the planning_scene_monitor, without attached objects support, but with use-case specific services/publishers for introspection.

**Comment by fmessmer** [comment]
Can we remove the auto-generated comments in package.xml and CMakeLists.txt?

Do we need a separate launch file in `cob_bringup` that uses this node?
~~There is no node yet that subscribes to the bool topic and stops the robot, right? Where will functionality be implemented?~~
Just saw the reference to `cob_safety_controller`

**Comment by mathias-luedtke** [comment]
> Do we need a separate launch file in cob_bringup that uses this node?

A launch file or at least a config file will be needed if the parameter defaults do not fit.

**Comment by mathias-luedtke** [comment]
I will rebase to #59 to get it tested before the merge.

**Comment by fmessmer** [comment]
I'm trying to get this to work...however, I'm not able to actually load the CollisionMonitor plugin when adding `cob_collision_monitor/CollisionMonitor` to the cob_moveit_config/move_group.launch.
The error I get during startup:

```
Loading 'cob_collision_monitor/CollisionMonitor'...
[ERROR] [1454691236.525031908, 31.606000000]: Exception while loading move_group capability 'cob_collision_monitor/CollisionMonitor': According to the loaded plugin descriptions the class cob_collision_monitor/CollisionMonitor with base class type move_group::MoveGroupCapability does not exist. Declared types are  cob_collision_monitor::CollisionMonitor move_group/ClearOctomapService move_group/MoveGroupCartesianPathService move_group/MoveGroupExecuteService move_group/MoveGroupGetPlanningSceneService move_group/MoveGroupKinematicsService move_group/MoveGroupMoveAction move_group/MoveGroupPickPlaceAction move_group/MoveGroupPlanService move_group/MoveGroupQueryPlannersService move_group/MoveGroupStateValidationService
Available capabilities: cob_collision_monitor::CollisionMonitor, move_group/ClearOctomapService, move_group/MoveGroupCartesianPathService, move_group/MoveGroupExecuteService, move_group/MoveGroupGetPlanningSceneService, move_group/MoveGroupKinematicsService, move_group/MoveGroupMoveAction, move_group/MoveGroupPickPlaceAction, move_group/MoveGroupPlanService, move_group/MoveGroupQueryPlannersService, move_group/MoveGroupStateValidationService
```
- I noticed the `::` and `/` discrepancy, thus, I added an explicit `name="cob_collision_monitor/CollisionMonitor"` to the cob_collision_monitor/move_group_capabilities.xml`. Did not help...
- I also exported the lib in `catkin_package()` within CMakeLists.txt...Did not help....
- I also tried switching to `class_loader` syntax
  
  ```
  #include <class_loader/class_loader.h>
  CLASS_LOADER_REGISTER_CLASS(cob_collision_monitor::CollisionMonitor, move_group::MoveGroupCapability)
  ```

I don't know what's wrong...@ipa-mdl, do you have an idea? Did you actually test it as capability within MoveIt! or just as node?

File: cob_collision_monitor/move_group_capabilities.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<library path="lib/libcob_collision_monitor_capabilty">
```
Comment (fmessmer) [inline cob_collision_monitor/move_group_capabilities.xml]: `<library path="lib/libcob_collision_monitor_capability">`

missing `i`!!!

**Comment by fmessmer** [comment]
remaining issues and rebase fixed in #61

**Comment by mathias-luedtke** [comment]
superseded -> close

**Pull request closed without merging**